plugins/modules/azure_rm_notificationhub.py

```diff
@@ -0,0 +1,360 @@
+__metaclass__ = type
+DOCUMENTATION = '''
+---
+module: azure_rm_notificationhub
+version_added: "0.1.2"
+short_description: Manage Notification Hub
+description:
+    - Create, update and delete instance of Notification Hub.
+options:
+    resource_groups:
+        description:
+            - Name of the resource group to which the resource belongs.
+    namespace_name:
+        description:
+            - Name of the namespace in which to create notification hub
+    name:
+        description:
+            - Unique name of the app service plan to create or update.
+        required: True
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+    sku:
+        description:
+            - The name of the SKU.
+            - Please see (https://azure.microsoft.com/en-in/pricing/details/notification-hubs/)
+        default: free
+        choices:
+            - free
+            - basic
+            - premium
+    state:
+      description:
+          - Assert the state of the Notification Hub.
+          - Use C(present) to create or update an notification hub and C(absent) to delete it.
+      default: present
+      choices:
+          - absent
+          - present
+extends_documentation_fragment:
+- azure.azcollection.azure
+
+author:
+    - Praveen Ghuge (@praveenghuge)
+    - Karl Dasan (@karldas30)
+'''
+EXAMPLES = '''
+
+- name: "Create Notification Hub"
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    name: myhub
+    tags:
+       - a: b
+    sku: free
+
+- name: Delete Notification Hub
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    name: myNamespace
+    state: absent
+
+- name: "Create Notification Hub Namespace"
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    location: eastus
+    namespace_name: myNamespace
+    tags:
+       - a: b
+    sku: free
+
+- name: Delete Notification Hub Namespace
+  azure_rm_notificationhub:
+    resource_group: testgroupans
+    namespace_name: myNamespace
+    state: absent
+
+'''
+
+RETURN = '''
+
+'''
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.notificationhubs.models import NotificationHubCreateOrUpdateParameters, NamespaceCreateOrUpdateParameters
+    from azure.mgmt.notificationhubs.models.sku import Sku
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureNotificationHub(AzureRMModuleBase):
+
+    def __init__(self):
+        # define user inputs from playbook
+
+        self.authorizations_spec = dict(
+            name=dict(type='str', required=True)
+        )
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            namespace_name=dict(type='str', required=True),
+            name=dict(type='str'),
```

Comment:
Argument 'name' in argument_spec is not required, but is documented as being required